curator-x-discovery/src/main/java/org/apache/curator/x/discovery/details/ServiceCacheImpl.java

```diff
@@ -78,7 +79,7 @@ private static ExecutorService convertThreadFactory(ThreadFactory threadFactory)
 
         this.discovery = discovery;
 
-        String path = discovery.pathForName(name);
+        path = discovery.pathForName(name);
```

Comment:
The bug might actually be in `forPathChildrenCache()` of `CuratorCacheListener`. We shouldn't have to filter the root out. I believe PathChildrenCache didn't report on the root so `forPathChildrenCache` should emulate that behavior.

Reply:
@Randgalt do you think this is a regression ?

I will push down the fix to CuratorCacheListener, thanks for the suggestion.
I didn't go that way because I wasn't sure.

It is important to understand if this is a regression in 5.0 and then we have to fix it asap, otherwise applications will break

Reply:
If pushing it down fixes it then it is a regression. I'm surprised we don't have a test for this. We can get a new version out if needed.

Reply:
I see this problem with ZK 3.5.7 not with 3.6.

btw @Randgalt  I see that CuratorCacheListener does not hold a reference to the root path, and it does not hold any reference to the CuratorCache.
I cannot introduce such reference, because it will be an incompatible change.

Reply:
We can see the problem even in ZK 3.6.1

Reply:
> I see this problem with ZK 3.5.7 not with 3.6.

Yeah - I'm mistaken. I thought it use PathChildrenCache in pre-3.6 but it uses TreeCache which will have the same problem.

Reply:
@eolivelli I think it could be handled by adding a new `CuratorCache.Options` - something like `IGNORE_ROOT` and then `CuratorCacheBridgeBuilder` would always set that option.